### PR TITLE
feature(azure): handle availability zone

### PR DESCRIPTION
**Pull request by soyacz**
SCT does not support availability zones for Azure backend.

This commit adds handling it for Azure (and unified provisioning).

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/provision/azure/provisioner.py [hunk 1]
```diff
@@ -34,38 +36,59 @@
...
         self._nic_provider = NetworkInterfaceProvider(self._resource_group_name, self._region, self._azure_service)
-        self._vm_provider = VirtualMachineProvider(self._resource_group_name, self._region, self._azure_service)
+        self._vm_provider = VirtualMachineProvider(
+            self._resource_group_name, self._region, self._az, self._azure_service)
         for v_m in self._vm_provider.list():
             self._cache[v_m.name] = self._vm_to_instance(v_m)
 
+    @staticmethod
+    def _convert_az_to_zone(availability_zone: str) -> str:
+        """When user provides letter instead of zone number, convert it to number."""
```
Comment (fgelcer) [inline sdcm/provision/azure/provisioner.py:62]: i don't fully understand what it means

File: test-cases/longevity/longevity-mini-test-1h.yaml [hunk 2]
```diff
@@ -13,4 +13,4 @@ nemesis_interval: 1
 user_prefix: 'longevity-mini-test'
 space_node_threshold: 6442
 
-append_scylla_args: '--memory 4G'
+#append_scylla_args: '--memory 4G'
```
Comment (fruch) [inline test-cases/longevity/longevity-mini-test-1h.yaml:16]: is this related to this PR ? (or intended ?)

Comment (fgelcer) [inline test-cases/longevity/longevity-mini-test-1h.yaml:16]: i think it should not be part of this PR, and shall be reverted

Comment (soyacz) [reply test-cases/longevity/longevity-mini-test-1h.yaml:16]: required for testing - as 4G memory is to little for 8 cores for Scylla (cannot create smaller machines).
I'd gladly increased this value to make this test work for Azure without this change.
For now I'll revert it back as it was just temporary change.

File: sdcm/provision/azure/provisioner.py (hunk 1 above)
Comment (soyacz) [reply sdcm/provision/azure/provisioner.py:62]: I added some more info why we need this function. If still it is not clear, let me know.

File: sdcm/provision/azure/provisioner.py [hunk 3]
```diff
@@ -34,38 +36,59 @@
...
+            return str(chars.find(availability_zone) + 1)
+        return availability_zone
+
+    @staticmethod
+    def _get_az_from_name(resource_group: ResourceGroup) -> str:
+        """Gets availability zone from name or return empty string if no az was used."""
+        # this method should be removed when all supported branches support az
+        name_last_two_letters = resource_group.name[-2:]
+        if name_last_two_letters[0] == "-":
+            return name_last_two_letters[1]
```
[the remark this answers is not in the document]
Comment (soyacz) [reply sdcm/provision/azure/provisioner.py:73]: fixed

**Review by fgelcer — APPROVED** [review]
LGTM

File: sdcm/cluster.py [hunk 4]
```diff
@@ -3860,7 +3860,7 @@ def get_nodetool_status(self, verification_node=None):  # pylint: disable=too-ma
         status = {}
         res = verification_node.run_nodetool('status', publish_event=False)
 
-        data_centers = res.stdout.strip().split("Datacenter: ")
```
Comment (fruch) [inline sdcm/cluster.py:3863]: why are we removing this `strip()` call ?

**Review by fruch — APPROVED** [review]
LGTM
